### DISCUSSION TITLE
Use Ray with a remote docker setup

### DISCUSSION THREAD

**Discussion by clementai** [spatie/laravel-ray#191] [Q&A] [answered]
Hello,

I'm developing on a Mac using VS Code with the remote SSH extension and a Ubuntu VPS running my app using docker (wow this is hard to explain).

I can't find a way to use Ray with that setup.
The Ray app is connected to my Ubuntu VPS using the SSH feature, I can verify that with the `lsof -i -P -n | grep LISTEN` command on the VPS, as you can see here:
![image](https://user-images.githubusercontent.com/6172122/117541827-084d8800-b016-11eb-940e-1e8e0a4eb593.png)

Next my Laravel app has the `laravel-ray` package installed with the `'host' => 'host.docker.internal'` config, and the `extra_hosts` set, I can ping `host.docker.internal:23517` from my Laravel container.

But even with all that using `ray('test')` from a controller or tinker doesn't work, nothing appear in the Ray app.

Am I missing something?

Thanks

**Comment by phamquocbuu** [comment]
Same problem here.
I use Ray app and VSCode in Mac OS >> remote Windows PC >> Docker ( my code )
-> Ray app cannot receive logs.

This case is test passed:
I use Ray app in Windows PC >> Docker ( my code )
-> Ray app can receive logs.

Both, Ray app SSH into the Docker container which containing my code.

**Comment by darman89** [comment, marked as answer]
In my Ubuntu VPS using the SSH feature, the SSH config tip in this [note](https://spatie.be/docs/ray/v1/the-ray-desktop-app/discovering-the-ray-app#connecting-to-remote-servers) worked for me:

> Note: if you are connecting to a docker container in a remote server (see [docker configuration](https://spatie.be/docs/ray/v1/environment-specific-configuration/docker)), you may need to enable `GatewayPorts yes` in server's `/etc/ssh/sshd_config` configuration (remember to restart the sshd daemon in order to apply your changes).

I hope it help you too.

P.D: sorry for my English
